apiml-security-common/src/main/java/org/zowe/apiml/security/common/auth/saf/SafResourceAccessConfig.java

```diff
@@ -10,34 +10,84 @@
 
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.web.client.RestTemplate;
+
 import java.io.IOException;
+import java.util.Arrays;
 
 @Configuration
 @Slf4j
 public class SafResourceAccessConfig {
 
-    @Value("${apiml.security.safEndpoint.enabled:false}")
+    private static final String ENDPOINT = "endpoint";
+    private static final String NATIVE = "native";
+    private static final String DUMMY = "dummy";
+
+    private static final String[] PROVIDERS = new String[] { ENDPOINT, NATIVE, DUMMY };
+
+    @Value("${apiml.security.authorization.provider:}")
+    private String provider;
+
+    @Value("${apiml.security.authorization.endpoint.enabled:false}")
     private boolean endpointEnabled;
 
-    @Bean
-    public SafResourceAccessVerifying safResourceAccessVerifying(RestTemplate restTemplate) throws IOException {
+    protected SafResourceAccessVerifying createEndpoint(RestTemplate restTemplate) {
+        return new SafResourceAccessEndpoint(restTemplate);
+    }
+
+    protected SafResourceAccessVerifying createNative() throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException, NoSuchFieldException {
+        return new SafResourceAccessSaf();
+    }
 
-        if (endpointEnabled) {
-            return new SafResourceAccessEndpoint(restTemplate);
+    protected SafResourceAccessVerifying createDummy() throws IOException {
+        return new SafResourceAccessDummy();
+    }
+
+    private SafResourceAccessVerifying create(RestTemplate restTemplate, String type, boolean force) {
+        switch (StringUtils.lowerCase(type)) {
+            case ENDPOINT:
+                if (endpointEnabled || force) {
+                    return createEndpoint(restTemplate);
+                }
+                return null;
+            case NATIVE:
+                try {
+                    return createNative();
+                } catch (Exception e) {
+                    log.debug("API PlatformAccessControl is not available", e);
+                }
+                return null;
+            case DUMMY:
+                try {
+                    return createDummy();
+                } catch (Exception e) {
+                    log.debug("saf.yml nor mock-saf.yml is not available", e);
+                }
+                return null;
+            default:
+                throw new IllegalArgumentException(
+                    String.format("Unknown provider to check SAF resource '%s', use one from %s",
+                       type, Arrays.toString(PROVIDERS)
+                    ));
         }
+    }
 
-        try {
-            return new SafResourceAccessSaf();
-        } catch (Exception e) {
-            log.debug("API PlatformAccessControl is not available", e);
+    @Bean
```

Comment:
Have you considered this approach to instantiating the Beans?

@ConditionalOnProperty(name = "caching.storage.mode", havingValue = "vsam")
    @Bean

Reply:
This would later on allow simplier adding of new mechanisms without touching existing code and modularization of these.

Reply:
Ok, I will think about, maybe next PR...